components/camel-kudu/src/main/java/org/apache/camel/component/kudu/KuduEndpoint.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.kudu;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+import org.apache.kudu.client.KuduClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Represents a Kudu endpoint. A kudu endpoint allows you to interact with
+ * <a href="https://kudu.apache.org/">Apache Kudu</a>, a free and open source
+ * column-oriented data store of the Apache Hadoop ecosystem.
+ */
+@UriEndpoint(firstVersion = "2.25.0",
+    scheme = "kudu",
+    title = "Apache Kudu", syntax = "kudu:host:port/tableName",
+    label = "cloud,database,iot", producerOnly = true)
+public class KuduEndpoint extends DefaultEndpoint {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KuduEndpoint.class);
+    private KuduClient kuduClient;
+
+    @Metadata(required = true, description = "Host of the server to connect to")
+    private String host;
+
+    @UriPath(description = "Connection string to Kudu")
+    private String uri;
+
+    @Metadata(required = true)
+    private String port;
+
+    @UriParam(description = "Operation to perform")
+    private KuduOperations operation;
+
+    @Metadata(description = "Table to connect to")
+    private String tableName;
+
+    public KuduEndpoint(String uri, KuduComponent component) {
+        super(uri, component);
+        this.setUri(uri);
+        this.setEndpointUri(uri);
```

Comment:
You don't need to set `this.setEndpointUri(uri);`, when you call the parent constructor of `DefaultEndpoint`, `EndpointUri` is being set there